File: composer/src/main/java/io/piano/android/composer/model/Event.java
package io.piano.android.composer.model;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;

import io.piano.android.composer.model.events.EventType;

@Keep
public class Event<T extends EventType> {
    @NonNull
    public EventModuleParams eventModuleParams;
    @NonNull
    public EventExecutionContext eventExecutionContext;
    @NonNull
    public T eventData;

    public Event(
            @NonNull EventModuleParams eventModuleParams,
            @NonNull EventExecutionContext eventExecutionContext,
            @NonNull T eventData
    ) {
        this.eventModuleParams = eventModuleParams;
        this.eventExecutionContext = eventExecutionContext;
        this.eventData = eventData;
    }
}

File: composer/src/main/java/io/piano/android/composer/model/Event.kt
package io.piano.android.composer.model

import io.piano.android.composer.model.events.EventType

data class Event<out T : EventType>(
    @JvmField val eventModuleParams: EventModuleParams,
    @JvmField val eventExecutionContext: EventExecutionContext,
    @JvmField val eventData: T
)
